src/protocols/aeternity/libs/AeAccountHdWallet.ts

```diff
@@ -0,0 +1,108 @@
+/* eslint-disable class-methods-use-this */
+import {
+  AccountBase,
+  sign,
+  messageToHash,
+  RpcRejectedByUserError,
+  unpackTx,
+  Encoded,
+  METHODS,
+  buildTx,
+  Tag,
+  decode,
+} from '@aeternity/aepp-sdk';
+import { Ref } from 'vue';
+import type { ITx } from '@/types';
+import { useAccounts } from '@/composables/accounts';
+import { usePermissions } from '@/composables/permissions';
+import {
+  IS_MOBILE_APP,
+  IS_EXTENSION_BACKGROUND,
+  PROTOCOL_AETERNITY,
+  IN_FRAME,
+} from '@/constants';
+
+interface InternalOptions {
+  fromAccount?: Encoded.AccountAddress;
+}
+
+export class AeAccountHdWallet extends AccountBase {
+  nodeNetworkId: Ref<string | undefined>;
+
+  constructor(nodeNetworkId: Ref<string | undefined>) {
+    super();
+    this.nodeNetworkId = nodeNetworkId;
+  }
+
+  override async signTransaction(
+    txBase64: Encoded.Transaction,
+    options: Parameters<AccountBase['signTransaction']>[1],
+  ): Promise<Encoded.Transaction> {
+    if (
+      (IS_MOBILE_APP || IS_EXTENSION_BACKGROUND || IN_FRAME)
+      && options?.aeppOrigin
+    ) {
+      const { checkOrAskPermission } = usePermissions();
+      const tx = unpackTx(txBase64) as any as ITx;
+      const hasPermission = await checkOrAskPermission(
+        options?.aeppOrigin,
+        METHODS.sign,
+        { ...options, txBase64, tx },
+      );
+      if (!hasPermission) {
+        throw new RpcRejectedByUserError('Rejected by user');
+      }
+    }
+
+    const encodedTx = decode(txBase64);
+    const signature = await this.sign(
+      Buffer.concat([
+        Buffer.from(this.nodeNetworkId.value!),
```

Comment:
Would it be safer if we throw an error if `this.nodeNetworkId.value` does not exist?

Reply:
Makes sense. I added a check for the `nodeNetworkId` and an error is thrown when it's missing.